nvme_gw_server.py

```diff
@@ -37,8 +38,11 @@ def start_spdk(self):
         spdk_tgt = self.nvme_config.get("config", "spdk_tgt")
         spdk_cmd = os.path.join(spdk_path, spdk_tgt)
         spdk_rpc_socket = self.nvme_config.get("spdk", "rpc_socket")
+        spdk_tgt_cmd_args = self.nvme_config.get("spdk", "tgt_cmd_args")
 
         cmd = [spdk_cmd, "-u", "-r", spdk_rpc_socket]
+        if spdk_tgt_cmd_args:
+            cmd += shlex.split(spdk_tgt_cmd_args)
```

Comment:
I'm not sure how spdk CLI parser behaves, but typically the latter arguments override the former.  Since `tgt_cmd_args` is essentially a free-form string which is appended after `cmd` is generated, it can potentially override the options generated from/by `[spdk]` section.  Is that desirable?  Should that be documented?

Reply:
I think it is desirable. I would say this is an "expert" option anyway and even if it does not override arguments can render SPDK unusable.

Reply:
Yes, in the spdk CLI parser the last argument wins. And yes, it was my intention that `tgt_cmd_args` would override `[spdk]` options. I though it would be more appropriate, because if user bothered to specify something in the "advanced" `tgt_cmd_args` option she would expect it was applied.

Now, thinking more about this, it might create a potential issue with `rpc_socket`. If a user by some reason decides to specify "--rpc-socket ..." in `tgt_cmd_args`, the target will use this option, but the nvme_gw_server will use `rpc_socket` option when trying to talk to the target.

I suppose we may not bother with issues like this, expecting from users they know what they do, but we may have some protection. E.g. we indeed, my change the order, so  `tgt_cmd_args` are overridden by `[spdk]` options. Or, more advanced, we my parse `tgt_cmd_args`, and disallow arguments that can be set via `[spdk]` options.

Reply:
I'm not sure if it's worth bothering about.  Like Jonas said, it's an "expert" option.  As long as we log the actual command that is invoked, an advanced user should be able to figure it out.